roxie/ccd/ccdprotocol.cpp

```diff
@@ -1837,26 +1837,39 @@ class RoxieSocketWorker : public ProtocolQueryWorker
                 if (httpHelper.isHttp())
                     httpHelper.setUseEnvelope(extractor.isSoap);
             }
+
+            if (!queryName)
+            {
+                // or should we set("") ?  Is an empty queryName ever valid ?
+                if (doTrace(traceSockets, TraceFlags::Max))
+                {
+                    StringBuffer b;
+                    DBGLOG("missing/invalid query name from socket");
+                }
+                client.clear();
+                return;
+            }
+
             if (streq(queryPrefix.str(), "control"))
             {
                 if (httpHelper.isHttp())
                     client->setHttpMode(queryName, false, httpHelper);
 
-                bool aclupdate = strieq(queryName, "aclupdate"); //ugly
+                bool aclupdate = strieq(queryName.str(), "aclupdate"); //ugly
```

Comment:
Not technically needed if we check for nullptr above, but to be fully correct code I added .str() to these